core/src/main/java/tc/oc/pgm/flag/state/Carried.java

```diff
@@ -213,7 +213,7 @@ public void tickRunning() {
     }
 
     if (!isFlag(carrier.getInventory().getHelmet())) {
-      this.dropFlag();
+      if (isCurrent()) this.dropFlag();
     }
```

Comment:
i find it somewhat concerning that `tickRunning` as a whole is called if its not the current state, how is this happening?